esp/services/ws_logaccess/WsLogAccessService.cpp

```diff
@@ -52,63 +52,145 @@ LogAccessTimeRange requestedRangeToLARange(IConstTimeRange & reqrange)
+       throw makeStringException(-1, "WsLogAccess: Empty searchByValue detected");
 
-    LogAccessConditions logFetchOptions;
     switch (searchByCategory)
     {
         case CLogAccessType_All:
-            logFetchOptions.setFilter(getWildCardLogAccessFilter());
-            break;
+            return getWildCardLogAccessFilter();
         case CLogAccessType_ByJobIdID:
-            logFetchOptions.setFilter(getJobIDLogAccessFilter(searchByValue));
-            break;
+            return getJobIDLogAccessFilter(searchByValue);
         case CLogAccessType_ByComponent:
-            logFetchOptions.setFilter(getComponentLogAccessFilter(searchByValue));
-            break;
+            return getComponentLogAccessFilter(searchByValue);
         case CLogAccessType_ByLogType:
         {
             LogMsgClass logType = LogMsgClassFromAbbrev(searchByValue);
             if (logType == MSGCLS_unknown)
                 throw makeStringExceptionV(-1, "Invalid Log Type 3-letter code encountered: '%s' - Available values: 'DIS,ERR,WRN,INF,PRO,MET'", searchByValue);
 
-            logFetchOptions.setFilter(getClassLogAccessFilter(logType));
-            break;
+            return getClassLogAccessFilter(logType);
         }
         case CLogAccessType_ByTargetAudience:
         {
             MessageAudience targetAud = LogMsgAudFromAbbrev(searchByValue);
             if (targetAud == MSGAUD_unknown || targetAud == MSGAUD_all)
                 throw makeStringExceptionV(-1, "Invalid Target Audience 3-letter code encountered: '%s' - Available values: 'OPR,USR,PRO,ADT'", searchByValue);
 
-            logFetchOptions.setFilter(getAudienceLogAccessFilter(targetAud));
-            break;
+            return getAudienceLogAccessFilter(targetAud);
         }
         case CLogAccessType_BySourceInstance:
         {
-            logFetchOptions.setFilter(getInstanceLogAccessFilter(searchByValue));
-            break;
+            return getInstanceLogAccessFilter(searchByValue);
         }
         case CLogAccessType_BySourceNode:
         {
-            logFetchOptions.setFilter(getHostLogAccessFilter(searchByValue));
+            return getHostLogAccessFilter(searchByValue);
             break;
         }
+        case CLogAccessType_ByFieldName:
+        {
+            return getColumnLogAccessFilter(serchField, searchByValue);
+        }
         case LogAccessType_Undefined:
         default:
             throw makeStringException(-1, "Invalid remote log access request type");
     }
+}
+
+ILogAccessFilter * processLogFilter(IConstLogFilter * logFilter)
+{
+    if (logFilter)
+        return processLogFilterByFields(logFilter->getLogCategory(), logFilter->getSearchByValue(), logFilter->getSearchField());
+    else
+        return nullptr;
+}
+
+ILogAccessFilter * processBinaryLogFilter(IConstBinaryLogFilter * binaryfilter)
+{
+    if (!binaryfilter)
+        return nullptr;
+    
+    ILogAccessFilter * leftFilter = nullptr;
```

Comment:
Should you give an error if binaryfilter->getLeftFilter() is filled in and binaryfilter->getLeftBinaryFilter().ordinality() != 0?

Since you don't support both together, checking would prevent user mistakes and confusion, and in some scenarios would let you change the behavior in the future without worrying about breaking existing users.

Reply:
If so, same would apply to the right hand side.

Reply:
Yep, good point.

Reply:
I think L146 tells you if no left filter was provided.  I meant the case where you have to many left filters because the nested one was filled in, and the array also had an entry.  Your functionality works and ignores the extra infor, but this would really be a check to make sure the caller knows what they are doing so if you make changes in the future the client won't break.

Reply:
Test case for scenario exposed by Tony, explicit error message now returned to user when left/rightFilter and left/rightBinaryFilter both submitted:

<urn:GetLogsRequest>
         <urn:Filter>
            <urn:leftFilter>
               <urn:LogCategory>1</urn:LogCategory>
               <urn:SearchByValue>W20220504-173939</urn:SearchByValue>
            </urn:leftFilter>
           
            <urn:Operator>1</urn:Operator>
                        <urn:rightFilter>
               <urn:LogCategory>1</urn:LogCategory>
               <urn:SearchByValue>W20220504-173939</urn:SearchByValue>
            </urn:rightFilter>
            <urn:rightBinaryFilter>
               <urn:BinaryLogFilter>
                  <urn:leftFilter>
                     <urn:LogCategory>3</urn:LogCategory>
                     <urn:SearchByValue>err</urn:SearchByValue>
                  </urn:leftFilter>
                  <urn:Operator>2</urn:Operator>
                  <urn:rightFilter>
                     <urn:LogCategory>3</urn:LogCategory>
                     <urn:SearchByValue>wrn</urn:SearchByValue>
                  </urn:rightFilter>
               </urn:BinaryLogFilter>
            </urn:rightBinaryFilter>
         </urn:Filter>
         <urn:Range>
            <urn:StartDate>2022-03-02T01:26:27</urn:StartDate>
         </urn:Range>
         <urn:LogLineLimit>10</urn:LogLineLimit>
         <urn:LogLineStartFrom>0</urn:LogLineStartFrom>
         <urn:SelectColumnMode>1</urn:SelectColumnMode>
         <urn:Format>XML</urn:Format>
      </urn:GetLogsRequest>